lucid/optvis/objectives.py

```diff
@@ -61,41 +61,30 @@ class Objective(object):
 
   def __init__(self, objective_func, name="", description=""):
     self.objective_func = objective_func
-    self.name = name
     self.description = description
+    self.value = None # This value is populated after a call
 
   def __add__(self, other):
     if isinstance(other, (int, float)):
       objective_func = lambda T: other + self(T)
-      name = self.name
-      description = self.description
     else:
       objective_func = lambda T: self(T) + other(T)
-      name = ", ".join([self.name, other.name])
-      description = "Sum(" + " +\n".join([self.description, other.description]) + ")"
-    return Objective(objective_func, name=name, description=description)
+    description = "(" + " + ".join([str(self), str(other)]) + ")"
+    return Objective(objective_func, description=description)
 
   def __neg__(self):
     return -1 * self
 
   def __sub__(self, other):
     return self + (-1 * other)
 
-  @staticmethod
-  def sum(objs):
-    objective_func = lambda T: sum([obj(T) for obj in objs])
-    descriptions = [obj.description for obj in objs]
-    description = "Sum(" + " +\n".join(descriptions) + ")"
-    names = [obj.name for obj in objs]
-    name = ", ".join(names)
-    return Objective(objective_func, name=name, description=description)
-
   def __mul__(self, other):
     if isinstance(other, (int, float)):
       objective_func = lambda T: other * self(T)
     else:
       objective_func = lambda T: self(T) * other(T)
-    return Objective(objective_func, name=self.name, description=self.description)
+    description = str(self) + "·" + str(other)
```

Comment:
I thought one needed to add something special to the header for python to be OK with unicode in the file? Maybe this was only an issue with older versions of python. Tests should catch it.

Reply:
You're right; that's [fine in Python 3](https://docs.python.org/3/howto/unicode.html#the-string-type), but Python 2 [asks you to declare](https://docs.python.org/2.7/howto/unicode.html#unicode-literals-in-python-source-code) an encoding.

Reply:
@gabgoh please add the following as the top line (Github won't let me propose it directly because you didn't change the first line… gaaaah code review tools.)
```
# -*- coding: utf-8 -*-
```